phpBB/phpbb/finder.php

```diff
@@ -421,10 +421,12 @@ public function find_from_extension($extension_name, $extension_path, $cache = t
 	*/
 	public function find_from_paths($extensions, $cache = true, $is_dir = false)
 	{
+		global $phpbb_container;
```

Comment:
I don't like this, I think it should be a parameter of the constructor but as a new finder is built each time, ww should create a finder_factory service that can be injected anywhere we need a finder.

Reply:
Yes, that is why i asked on irc

Reply:
But... i dont like this finder that do a lot of things, i think there should be a finder, and other finder that extends from this one or something in extensions folder

Reply:
don't think it's a good idea to have multiple classes. The whole point of the finder is to be configurable.